generate_polices.go

```diff
@@ -0,0 +1,117 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+    "fmt"
+    "bytes"
+)
+
+type ruleOption struct {
+    occurance int
+    g         generator
+}
+
+func generateAuthorizationPolicy(action string, ruleToOccurences map[string]*ruleOption) (string, error) {
+    yaml := bytes.Buffer{}
+    yaml.WriteString("spec:\n")
```

Comment:
The `spec` part should be generated from the protobuf API (`"istio.io/api/security/v1beta1"`), it's probably fine for now for a quick test but you will need to change it before sending them to the official github repo.

So basically you will need to import the protobuf API `"istio.io/api/security/v1beta1"` and manipulate the protobuf message directly, and then convert it to the yaml output automatically (see  https://github.com/istio/istio/blob/6d0953f791f76f614c2478f463eef166ea399105/pkg/util/protomarshal/protomarshal.go#L46).

Reply:
Sounds good I will create that in a separate pull request

Reply:
you can just copy that ToYAML() implementation if needed.